tamr_unify_client/dataset/resource.py

```diff
@@ -247,6 +248,23 @@ def from_geo_features(self, features, geo_attr=None):
             self._features_to_updates(features, record_id, key_attrs, geo_attr)
         )
 
+    def upstream_datasets(self):
+        """The Dataset's upstream datasets.
+
+        API returns the URIs of the upstream datasets,
+        resulting in a list of DatasetURIs, not actual Datasets.
+
+        :return: A list of the Dataset's upstream datasets.
+        :rtype: list[:class:`~tamr_unify_client.dataset.uri.DatasetURI`]
+        """
+        upstream_list = []
+        alias = self.api_path + "/upstreamDatasets"
+        resources = self.client.get(alias).successful().json()
+
+        for uri in resources:
+            upstream_list.append(DatasetURI(self.client, uri))
+        return upstream_list
```

Comment:
you could replace `upstream_list` and its computation by:

```python
return [DatasetURI(self.client, uri) for uri in resources]
```

^this uses a [list comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)